submariner-operator/issue-79-kubeconfig-handling.md

```diff
@@ -0,0 +1,101 @@
+## Epic Description
+
+The various `subctl` commands support a mixture of `KUBECONFIG` environment variables, `--kubeconfig` flags, `--kubecontext` flags,
+kubeconfig files etc.
+
+We should make this consistent, ideally using `clientcmd`’s tools to handle kubeconfig.
+[Operator PR #23](https://github.com/skitt/submariner-operator/pull/23) has an old attempt at this,
+using [`BindOverrideFlags()`](https://pkg.go.dev/k8s.io/client-go@v0.23.5/tools/clientcmd#BindOverrideFlags) to provide access
+to all the features supported by `clientcmd` (see `kubectl options`).
+
+## Acceptance Criteria
+
+For commands using a single context, _all_ the following should work,
+with kubeconfigs containing one or more clusters and/or contexts (this uses a `kube` prefix when setting up `clientcmd`):
+
+```bash
+# Use the default context in the kubeconfig
+KUBECONFIG=/path/to/kubeconfig subctl foo
+subctl foo --kubeconfig /path/to/kubeconfig
```

Comment:
In the current implementation when I have a kubeconfig with multiple contexts and use `subctl gather ...` command, it executes the command for each context. What would be the behaviour with for such commands according to this proposal?

Reply:
See the updated proposal.